Library/Homebrew/cmd/formulae.sh

```diff
@@ -16,9 +16,9 @@ homebrew-formulae() {
            -name vendor \
           \) \
          -prune -false -o -name '*\.rb' | \
-    sed -r -e 's/\.rb//g' \
-           -e 's_.*/Taps/(.*)/(home|linux)brew-_\1/_' \
-           -e 's|/Formula/|/|' \
+    sed -e 's/\.rb//g' \
+        -e 's_.*/Taps/\(.*\)/\(home\|linux\)brew-_\1/_' \
+        -e 's|/Formula/|/|' \
```

Comment:
```suggestion
    sed -e 's:.*/Taps/\([^/]*\)/[homelinux]*brew-\([^/]*\).*/\(.*\).rb:\1/\2/\3:' \
```
The alternation operator isn't part of [POSIX BRE](https://en.wikipedia.org/wiki/Regular_expression#POSIX_basic_and_extended), but we can get close to it using a bracket regular expression. We can also perform all three steps with a single expression, for which I used the colon as a delimiter since it can't be used in macOS paths.

Reply:
> The alternation operator isn't part of [POSIX BRE](https://en.wikipedia.org/wiki/Regular_expression#POSIX_basic_and_extended)

😢 

We can limit `[homelinux]*` to`[homelinux]\+` or `[homelinux]\{4,5\}` -- what do you think?

Reply:
The `+` isn't part of BRE either, but `[homelinux]\{4,5\}` will work.

Reply:
I like `[hl][oi][mn][eu]x*` :)
Though... I wonder if it would be easier to detect the right flag (-E vs -r).

Reply:
Or: use `awk` instead?

Reply:
There's no backreferences in `awk` (except for `gawk`'s `gensub()`), but this will work instead of `sed`:
```
awk '{sub(".*/Taps/","");sub("/(home|linux)brew-","/");sub("/Formula/","/");sub(".rb","");print}'
```